### PR TITLE
ElectricityMaps regional settings bug fix

### LINKED ISSUE #227: [Bug]:  ElectricityMapsDataSource fails with regional settings with comma as decimal separator
### Contact Details

_No response_

### What happened?

Running the CA-SDK with regional settings e.g. Danish that uses comma as decimal separator fails. The HTTP request to ElectricityMaps API urlencodes the comma to %2C

### client

WebAPI (Default)

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by pritipath**
Issue Number: #227 

# DEPENDENT PR
> **Warning**
> This PR depends on commits in other PR
> **DO NOT** merge this PR until  #203 merged and this branch is updated against the latest dev branch

## Summary
Fixes bug where ElectricityMapsDataSource fails with regional settings with comma as decimal separator

## Changes

- List of comprehensive changes

## Checklist

- [ ] Local Tests Passing?
- [ ] CICD and Pipeline Tests Passing?
- [ ] Added any new Tests?
- [ ] Documentation Updates Made?
- [ ] Are there any API Changes? If yes, please describe below.
- [ ] This is not a breaking change. If it is, please describe it below.

## Are there API Changes?
If yes, what are the expected API Changes? Please link to an API-Comparison workflow with the API Diff.

## Is this a breaking change?
If yes, what workflow does this break? 

## Anything else?
Other comments, collaborators, etc.
> Please follow [GitHub's suggested syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to link Pull Requests to Issues via keywords

This PR Closes #227

### CODE REVIEW

**Review by gfmatthews — APPROVED** [review]
Looks good to me - thanks for jumping on this so quickly!

**Review by Lybecker — APPROVED** [review]

**Comment by gfmatthews** [comment]
@vaughanknight / @Willmish - This is ready for review

**Review by bderusha — APPROVED** [review]

File: src/CarbonAware/test/model/LocationTests.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿using CarbonAware.Model;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CarbonAware.Tests.model;
+
```
Comment (juzuluag) [inline src/CarbonAware/test/model/LocationTests.cs:10]: nit:
```suggestion
[TestFixture]
```

File: src/CarbonAware/test/model/LocationTests.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+public class LocationTests
+{
+    private static Location _defaultLocation = new Location() { Name = "eastus", Latitude = 34.123m, Longitude = 123.456m };
+
+    [Test]
```
Comment (juzuluag) [inline src/CarbonAware/test/model/LocationTests.cs:15]: qq: Shouldn't be this simplified using only one method with `[TestCase(....)]` attribute?

**Review by juzuluag — COMMENTED** [review]
One comment. Other than that lgmt 👍

**Review by vaughanknight — APPROVED** [review]
Looks good, this was something that came up in the hackathon so it's great to see a fix come through.  Thanks again!

**Pull request merged**